src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/config/WebClientConfiguration.kt

```diff
@@ -64,55 +67,55 @@ class WebClientConfiguration(
-    return getWebClientCredentials(authorizedClientManager, prisonerSearchRootUri)
+    return getWebClientCredentials(authorizedClientManager, prisonerSearchRootUri, observationRegistry)
   }
 
   @Bean
   fun arnWebClientClientCredentials(authorizedClientManager: OAuth2AuthorizedClientManager): WebClient {
-    return getWebClientCredentials(authorizedClientManager, arnRootUri)
+    return getWebClientCredentials(authorizedClientManager, arnRootUri, observationRegistry)
   }
 
   @Bean
   fun prisonWebClientCredentials(authorizedClientManager: OAuth2AuthorizedClientManager): WebClient {
-    return getWebClientCredentials(authorizedClientManager, prisonRootUri)
+    return getWebClientCredentials(authorizedClientManager, prisonRootUri, observationRegistry)
   }
 
   @Bean
   fun keyWorkerWebClientCredentials(authorizedClientManager: OAuth2AuthorizedClientManager): WebClient {
-    return getWebClientCredentials(authorizedClientManager, keyWorkerRootUri)
+    return getWebClientCredentials(authorizedClientManager, keyWorkerRootUri, observationRegistry)
   }
 
   @Bean
   fun caseNotesWebClientCredentials(authorizedClientManager: OAuth2AuthorizedClientManager): WebClient {
-    return getWebClientCredentials(authorizedClientManager, caseNotesRootUri)
+    return getWebClientCredentials(authorizedClientManager, caseNotesRootUri, observationRegistry)
   }
 
   @Bean
   fun allocationManagerWebClientCredentials(authorizedClientManager: OAuth2AuthorizedClientManager): WebClient {
-    return getWebClientCredentials(authorizedClientManager, allocationManagerRootUri)
+    return getWebClientCredentials(authorizedClientManager, allocationManagerRootUri, observationRegistry)
   }
 
   @Bean
   fun rpDeliusWebClientCredentials(authorizedClientManager: OAuth2AuthorizedClientManager): WebClient {
-    return getWebClientCredentials(authorizedClientManager, rpDeliusRootUri)
+    return getWebClientCredentials(authorizedClientManager, rpDeliusRootUri, observationRegistry)
   }
 
   @Bean
   fun educationEmploymentWebClientCredentials(authorizedClientManager: OAuth2AuthorizedClientManager): WebClient {
-    return getWebClientCredentials(authorizedClientManager, educationEmploymentRootUri)
+    return getWebClientCredentials(authorizedClientManager, educationEmploymentRootUri, observationRegistry)
   }
 
   @Bean
   fun interventionsWebClientCredentials(authorizedClientManager: OAuth2AuthorizedClientManager): WebClient {
-    return getWebClientCredentials(authorizedClientManager, interventionsRootUri)
+    return getWebClientCredentials(authorizedClientManager, interventionsRootUri, observationRegistry)
   }
 
-  private fun getWebClientCredentials(authorizedClientManager: OAuth2AuthorizedClientManager, baseUrl: String): WebClient {
+  private fun getWebClientCredentials(authorizedClientManager: OAuth2AuthorizedClientManager, baseUrl: String, observationRegistry: ObservationRegistry): WebClient {
```

Comment:
You don't need to pass this through as a parameter - you can just use the field directly here

Reply:
Done